src/Contract/Types/Plutus/BidTerms.purs

```diff
@@ -49,3 +60,48 @@ instance FromData BidTerms where
     | n == BigNum.zero && recLength (Proxy :: Proxy BidTerms) == length pd =
         wrap <$> fromDataRec bidTermsSchema pd
   fromData _ = Nothing
+
+--------------------------------------------------------------------------------
+-- Validation
+--------------------------------------------------------------------------------
+
+validateBidTerms :: CurrencySymbol -> AuctionTerms -> BidTerms -> Effect Boolean
+validateBidTerms auctionCs (AuctionTerms auctionTerms) (BidTerms bidTerms) =
+  conj <$> verifySellerSignature <*> verifyBidderSignature
+  where
+  bidderInfo = unwrap bidTerms.bidder
+
+  verifyBidderSignature :: Effect Boolean
+  verifyBidderSignature =
+    case bidderInfo.bidderAddress of
+      bidderAddress | Just bidderPkh <- toPubKeyHash bidderAddress -> do
+        let payload = bidderSignatureMessage auctionCs bidderPkh bidTerms.price
+        sigStruct <- mkSigStructure bidderAddress payload
+        verifySignature bidderInfo.bidderVk sigStruct bidTerms.bidderSignature
+      _ -> pure false
+
+  verifySellerSignature :: Effect Boolean
+  verifySellerSignature = do
+    let payload = sellerSignatureMessage auctionCs bidderInfo.bidderVk
+    sigStruct <- mkSigStructure auctionTerms.sellerAddress payload
+    verifySignature auctionTerms.sellerVk sigStruct bidTerms.sellerSignature
+
+bidderSignatureMessageSize :: Int
+bidderSignatureMessageSize = 69
```

Comment:
Magic numbers should have explanatory notes in code comments, or an unbroken link to docs.

Reply:
Addressed in https://github.com/mlabs-haskell/hydra-auction-offchain/pull/15/commits/9f8727c47a12d196a779a244298f79a984200af9